src/cpp/src/speculative_decoding/speculative_decoding_stateful_eagle3.cpp

```diff
@@ -0,0 +1,1242 @@
+
+    std::ostringstream ss;
+    ss << "[EAGLE3-TENSOR] " << name << " values (" << elements_to_show << " elements): ";
+
+    const auto elem_type = tensor.get_element_type();
+    if (elem_type == ov::element::i64) {
+        const int64_t* data = tensor.data<const int64_t>();
+        for (size_t i = 0; i < elements_to_show; ++i) {
+            if (i > 0)
+                ss << ", ";
+            ss << data[i];
+        }
+    } else if (elem_type == ov::element::f32) {
+        const float* data = tensor.data<const float>();
+        for (size_t i = 0; i < elements_to_show; ++i) {
+            if (i > 0)
+                ss << ", ";
+            ss << std::fixed << std::setprecision(4) << data[i];
+        }
+    } else if (elem_type == ov::element::i32) {
+        const int32_t* data = tensor.data<const int32_t>();
+        for (size_t i = 0; i < elements_to_show; ++i) {
+            if (i > 0)
+                ss << ", ";
+            ss << data[i];
+        }
+    }
+
+    if (!show_all && elements_to_show < total_elements) {
+        ss << " ... (+" << (total_elements - elements_to_show) << " more)";
+    }
+    std::cout << ss.str() << std::endl;
+}
+
+void Eagle3InferWrapperBase::log_model_inputs(const ov::Tensor& input_ids,
+                                              const ov::Tensor& attention_mask,
+                                              const ov::Tensor& position_ids) const {
+    if (!m_verbose)
+        return;
+
+    std::cout << "[EAGLE3-INPUT] ========== MODEL INPUTS ==========" << std::endl;
+    log_tensor_info("input_ids", input_ids);
+    log_tensor_content("input_ids", input_ids, 0);
+    log_tensor_info("attention_mask", attention_mask);
+    log_tensor_content("attention_mask", attention_mask, 0);
+    log_tensor_info("position_ids", position_ids);
+    log_tensor_content("position_ids", position_ids, 0);
+    std::cout << "[EAGLE3-INPUT] =================================" << std::endl;
+}
+
+void Eagle3InferWrapperBase::log_model_outputs(const ov::Tensor& logits, const ov::Tensor& hidden_features) const {
+    if (!m_verbose)
+        return;
+
+    std::cout << "[EAGLE3-OUTPUT] ========== MODEL OUTPUTS =========" << std::endl;
+
+    if (logits && logits.get_size() > 0) {
+        const auto logits_shape = logits.get_shape();
+        log_tensor_info("logits", logits);
+
+        if (logits_shape.size() == 3) {
+            const size_t seq_len = logits_shape[1];
+            const size_t vocab_size = logits_shape[2];
+
+            // Show only last few positions for long sequences
+            const size_t start_pos = (seq_len > 5) ? (seq_len - 5) : 0;
+            if (start_pos > 0) {
+                std::cout << "[EAGLE3-OUTPUT] Showing last " << (seq_len - start_pos)
+                          << " positions (total: " << seq_len << ")" << std::endl;
+            }
+
+            // Show top-k logits for each position
```

Comment:
We don't need top-k for now... And this is way too much debug utilities in Eagle3 speculative decode file. I suggest to move all these utilities into another `eagle3_debug_utils.cpp` file and align it with  [src/cpp/src/debug_utils.hpp](https://github.com/openvinotoolkit/openvino.genai/blob/master/src/cpp/src/debug_utils.hpp#L12) file.